Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -2514,3 +2588,66 @@ void vtkSegmentation::CopySegment(vtkSegment* destination, vtkSegment* source, v
     }
   }
 }
+
+//----------------------------------------------------------------------------
+void vtkSegmentation::classInitialize()
+{
+  // Allocate the singleton
+  vtkSegmentationRandomSequence = vtkSegmentation::GetSegmentIDRandomSequenceInstance();
+}
+
+//----------------------------------------------------------------------------
+void vtkSegmentation::classFinalize()
+{
+  vtkSegmentationRandomSequence->Delete();
+  vtkSegmentationRandomSequence = nullptr;
+}
+
+//----------------------------------------------------------------------------
+// Return the single instance of vtkMinimalStandardRandomSequence
+vtkMinimalStandardRandomSequence* vtkSegmentation::GetSegmentIDRandomSequenceInstance()
+{
+  if (!vtkSegmentationRandomSequence)
+  {
+    // Try the factory first
+    vtkSegmentationRandomSequence = (vtkMinimalStandardRandomSequence*)vtkObjectFactory::CreateInstance("vtkMinimalStandardRandomSequence");
+    // if the factory did not provide one, then create it here
+    if (!vtkSegmentationRandomSequence)
+    {
+      vtkSegmentationRandomSequence = vtkMinimalStandardRandomSequence::New();
+    }
+
+    // Get the current ms since the epoch
+    time_t currentTime = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+    int seed = currentTime % 2147483646;
+    vtkSegmentationRandomSequence->SetSeed(currentTime);
```

Comment:
Do we really need to seed every time? (I would have assumed that we only need to set the seed when a new vtkSegmentationRandomSequence is created)

Reply:
It was only set once when the singleton hadn't been defined yet. In the new version this is explicit as it is in classInitialize().